src/schedule/ll_schedule.c

```diff
@@ -177,8 +176,8 @@ static void schedule_ll_client_reschedule(struct ll_schedule_data *sch)
 	/* rearm only if there is work to do */
 	if (atomic_read(&sch->domain->total_num_tasks)) {
 		/* traverse to set timer according to the earliest task */
-		list_for_item_safe(wlist, tlist, &sch->tasks) {
-			task = container_of(wlist, struct task, list);
+		list_for_item(tlist, &sch->tasks) {
```

Comment:
Can anyone else modify the list here ? i.e. are IRQs OFF

Reply:
`list_for_item_safe()` doesn't protect against others modifying the list, it only protects against the list being modified in the loop.